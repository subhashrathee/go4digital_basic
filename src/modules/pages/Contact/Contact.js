import React from 'react'
import HtmlToReact from 'html-to-react'
import Body from '../../common/Body/Body'
import Footer from '../../common/Footer/Footer'
import ContactUsForm from '../../common/ContactUsForm/ContactUsForm'
import JSONContent from '../../config/content/contact_us.js' 

//Content passed into body component
var content = "";

//Content to be used for google maps
const content_google_map = JSONContent.contact_us_google_map.join('');


/**
* Contact us page component used to render the page.
* This makes use of the contactUsForm component.
*
* Props:      None
* 
* State vars: None
*/
class Contact extends React.Component {
  render() {

    //given we're parsing html content, we'll use this now
    var HtmlToReactParser = HtmlToReact.Parser;
    var htmlToReactParser = new HtmlToReactParser();
    var map_content = htmlToReactParser.parse(content_google_map);

    return (
      <div>
        <div className="hero-image-home"></div>
        <Body heading="Contact Us" content={content}/>
        <ContactUsForm />
       
        <div className="main">
        <h1>Our address</h1>
        <p>3/14 Yarrbat Ave VIC 3103</p>
      
        {map_content}
  
        </div>

        <Footer />
      </div>
    )
  }
}

export default Contact;
