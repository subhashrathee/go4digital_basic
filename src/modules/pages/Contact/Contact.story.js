import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { BrowserRouter } from 'react-router-dom'
import Contact from './Contact'

storiesOf('Pages', module)
  .add('Contact', () => (
      <Contact />
  ));
