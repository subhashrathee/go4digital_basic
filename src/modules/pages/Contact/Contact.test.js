import React from 'react';
import Contact from './Contact';
import renderer from 'react-test-renderer';
import { BrowserRouter, Link } from 'react-router-dom'

test('Contact component', () => {
  const component = renderer.create(
    <Contact />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
