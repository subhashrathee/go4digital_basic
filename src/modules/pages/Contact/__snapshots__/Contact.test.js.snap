// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component 1`] = `
<div>
  <div
    className="hero-image-contact"
  />
  <div
    className="main"
  >
    <h1>
      Contact Us
    </h1>
  </div>
  <div
    className="main"
  >
    <div
      id="thankYouMessage"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <p>
        Thanks for submitting your enquiry. We'll get back to you shortly
      </p>
    </div>
    <form
      id="form"
    >
      <p>
        Call Sarita Singh for any enquiries on: 0414 043 200
      </p>
      <label
        id="emailLabel"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email address"
        required={true}
        type="email"
        value=""
      />
      <div
        className="error"
        id="emailError"
        style={
          Object {
            "color": "red",
            "fontSize": "12px",
            "marginTop": "5px",
          }
        }
      >
        
      </div>
      <br />
      <label
        id="messageLabel"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Message
      </label>
      <textarea
        cols="30"
        id="message"
        name="message"
        onChange={[Function]}
        placeholder="Message"
        required={true}
        rows="5"
        value=""
      />
      <div
        className="error"
        id="messageError"
        style={
          Object {
            "color": "red",
            "fontSize": "12px",
            "marginTop": "5px",
          }
        }
      >
        
      </div>
      <br />
      <label
        id="forbiddenInputLabel"
        style={
          Object {
            "display": "none",
          }
        }
      />
      <input
        id="forbiddenInput"
        name="forbiddenInput"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="input"
        value=""
      />
      <input
        name="submit"
        onClick={[Function]}
        type="submit"
        value="Submit"
      />
    </form>
  </div>
  <div
    className="main"
  >
    <h1>
      Our address
    </h1>
    <p>
      3/14 Yarrbat Ave VIC 3103
    </p>
    <iframe
      frameBorder="0"
      height="500"
      marginHeight="0"
      marginWidth="0"
      scrolling="no"
      src="https://maps.google.com.au/maps?q=3%2F14%20yarrbat%20ave%20balwyn&t=&z=15&ie=UTF8&iwloc=&output=embed"
      width="800"
    />
  </div>
  <footer>
    <p>
      Â© 
      2018
       go4digital. All rights reserved.
    </p>
  </footer>
</div>
`;
